packages/strapi-parts/src/MainNav/NavLink.js

```diff
@@ -57,8 +58,27 @@ const MainNavRow = styled(Row)`
   padding: ${({ theme }) => `${theme.spaces[2]} ${theme.spaces[3]}`};
 `;
 
-export const NavLink = ({ children, icon, ...props }) => {
+const Notification = styled(Row)`
+  //positionning when condensed
+  position: ${(props) => (props.condensed ? 'absolute' : '')};
+  transform: ${(props) => (props.condensed ? 'translate(35%, -50%)' : '')};
+  top: ${(props) => (props.condensed ? '0' : '')};
+  right: ${(props) => (props.condensed ? '0' : '')};
+
+  background: ${({ theme }) => theme.colors.primary600};
+  height: ${({ theme }) => theme.spaces[5]};
+  padding: ${({ theme }) => `0 ${theme.spaces[2]}`};
+  border-radius: ${({ theme }) => theme.spaces[10]};
+
+  ${Text} {
+    color: ${({ theme }) => theme.colors.neutral0} !important;
+    line-height: 0;
+  }
+`;
```

Comment:
Can you use the existing Badge component please and a badge wrapper.
```suggestion
const BadgeWrapper = styled(Row)`
  //positionning when condensed
  ${({ condensed }) =>  condensed && `
	position: absolute;
	transform: translate(35%, -50%);
	top: 0;
	right: 0;
   `}

  ${Text} {
    color: ${({ theme }) => theme.colors.neutral0} !important;
  }
`;
const CustomBadge = styled(Badge)`
  height: ${({ theme }) => theme.spaces[5]};
  padding: ${({ theme }) => `0 ${theme.spaces[2]}`};
  border-radius: ${({ theme }) => theme.spaces[10]};
`;
```

Reply:
Can you also add a comment above the !important please. We need to find a solution to avoid that bad practice.